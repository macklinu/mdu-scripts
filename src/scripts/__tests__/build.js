const cases = require('jest-in-case')
const {
  unquoteSerializer,
  winPathSerializer,
} = require('./helpers/serializers')

expect.addSnapshotSerializer(unquoteSerializer)
expect.addSnapshotSerializer(winPathSerializer)
expect.addSnapshotSerializer({
  print: val => val.replace(process.cwd(), 'PROJECT_ROOT'),
  test: val => typeof val === 'string',
})

jest.mock('rimraf', () => ({
  sync: () => {},
}))

cases(
  'build',
  ({ args = [] }) => {
    // beforeEach
    const mockExeca = require('execa')
    const utils = require('../../utils')
    utils.resolveBin = (modName, { executable = modName } = {}) => executable
    process.exit = jest.fn()

    // tests
    process.argv = ['node', '../build', ...args]
    mockExeca.mockClear()
    require('../build')
    expect(mockExeca).toHaveBeenCalledTimes(1)
    const [firstCall] = mockExeca.mock.calls
    const [script, calledArgs] = firstCall
    expect([script, ...calledArgs].join(' ')).toMatchSnapshot()

    // afterEach
    jest.resetModules()
  },
  {
    'calls babel with default args': {},
    'honors custom config': {
      args: ['--config', './my-config.js'],
    },
    'honors custom ignore': {
      args: ['--ignore', '__someDir__'],
    },
    'honors --no-copy-files': {
      args: ['--no-copy-files'],
    },
    'honors custom out dir': {
      args: ['--out-dir', 'destination'],
    },
  }
)
