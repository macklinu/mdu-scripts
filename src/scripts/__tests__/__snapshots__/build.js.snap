// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build calls babel with default args 1`] = `babel --out-dir dist --copy-files --ignore __tests__,__mocks__,**/*.test.js --presets PROJECT_ROOT/src/config/babelrc.js src`;

exports[`build honors --no-copy-files 1`] = `babel --out-dir dist --ignore __tests__,__mocks__,**/*.test.js --presets PROJECT_ROOT/src/config/babelrc.js src --no-copy-files`;

exports[`build honors custom config 1`] = `babel --out-dir dist --copy-files --ignore __tests__,__mocks__,**/*.test.js --presets PROJECT_ROOT/src/config/babelrc.js src --config ./my-config.js`;

exports[`build honors custom ignore 1`] = `babel --out-dir dist --copy-files --presets PROJECT_ROOT/src/config/babelrc.js src --ignore __someDir__`;

exports[`build honors custom out dir 1`] = `babel --copy-files --ignore __tests__,__mocks__,**/*.test.js --presets PROJECT_ROOT/src/config/babelrc.js src --out-dir destination`;
